pkg/comp-functions/functions/vshnmariadb/mariadb_deploy.go

```diff
@@ -396,3 +411,64 @@ func getReleaseSecretValue(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNMariaDB
 	}
 	return m["release"]
 }
+
+// To make scaling as seamless as possible we create an additional service.
+// This service will either point to a single db instance, or to the proxysql cluster.
+func createMainService(comp *vshnv1.VSHNMariaDB, svc *runtime.ServiceRuntime, secretName string) error {
```

Comment:
Technically speaking the customer can also use the svc directly from galera cluster bypassing proxysql, is that right?

Reply:
Yes, netpols only apply to pods and not the svcs... so there's no easy way to block this.
https://kubernetes.io/docs/concepts/services-networking/network-policies/#what-you-can-t-do-with-network-policies-at-least-not-yet

But on the other hand, it's probably only a question of time until we find the first app that doesn't like proxysql, just like with pgbouncer, so having a direct option for that might be a good thing.